streams/src/main/java/org/apache/kafka/streams/kstream/internals/TimeWindowedKStreamImpl.java

```diff
@@ -224,12 +225,25 @@
                         + " retention=[" + retentionPeriod + "]");
             }
 
-            supplier = Stores.persistentTimestampedWindowStore(
+            // get from default store implementation
+            final StoreImplementation storeImplementation = materialized.storeImplementation();
+            if (storeImplementation != null) {
+                supplier = storeImplementation.windowBytesStoreSupplier(
+                    materialized.storeName(),
+                    Duration.ofMillis(retentionPeriod),
+                    Duration.ofMillis(windows.size()),
+                    false);
+            } else {
```

Comment:
Also since we always default to the `persistent` type at the MaterializedInternal constructor if the config is not set, we should never need to hit this fallback logic right? If I read it wrong, could we make sure that at Materialized constructor if the config value is not set (should not be the case since we define its default value indeed?), we then hard-code a persistent type.